clang/lib/Sema/SemaChecking.cpp

```diff
@@ -4461,6 +4462,23 @@ bool Sema::CheckRISCVBuiltinFunctionCall(const TargetInfo &TI,
   return false;
 }
 
+bool Sema::CheckRISCVCOREVBuiltinFunctionCall(unsigned BuiltinID,
+                                           CallExpr *TheCall) {
+  // For intrinsics which take an immediate value as part of the instruction,
+  // range check them here.
+  switch (BuiltinID) {
+  default: return false;
+  case RISCVCOREV::BI__builtin_riscv_cv_simd_extract_h:
+  case RISCVCOREV::BI__builtin_riscv_cv_simd_extract_b:
+  case RISCVCOREV::BI__builtin_riscv_cv_simd_extractu_h:
+  case RISCVCOREV::BI__builtin_riscv_cv_simd_extractu_b:
+    return SemaBuiltinConstantArgRange(TheCall, 1, 0, 63);
+  case RISCVCOREV::BI__builtin_riscv_cv_simd_insert_h:
+  case RISCVCOREV::BI__builtin_riscv_cv_simd_insert_b:
+    return SemaBuiltinConstantArgRange(TheCall, 2, 0, 63);
```

Comment:
Same fix as above for the range

Reply:
I have fixed the ranges accordingly. I just noticed that I have run clang-format on the files. Is it fine or should I reverse it?